src/services/codefixes/returnValueCorrect.ts

```diff
@@ -0,0 +1,211 @@
+    const errorCodes = [
+        Diagnostics.A_function_whose_declared_type_is_neither_void_nor_any_must_return_a_value.code,
+        Diagnostics.Type_0_is_not_assignable_to_type_1.code,
+        Diagnostics.Argument_of_type_0_is_not_assignable_to_parameter_of_type_1.code
+    ];
+
+    enum FixKind {
+        MissingReturnStatement,
+        MissingParentheses
+    }
+
+    interface MissingReturnInfo {
+        kind: FixKind.MissingReturnStatement;
+        declaration: FunctionLikeDeclaration;
+        expression: Expression;
+        statement: Statement;
+    }
+
+    interface MissingParenInfo {
+        kind: FixKind.MissingParentheses;
+        declaration: ArrowFunction;
+        expression: Expression;
+        statement: Statement;
+    }
+
+    type Info = MissingReturnInfo | MissingParenInfo;
+
+    registerCodeFix({
+        errorCodes,
+        fixIds: [fixIdAddReturnStatement, fixIdRemoveBlockBodyBrace, fixIdWrapTheBlockWithParen],
+        getCodeActions: context => {
+            const { program, sourceFile, span: { start }, errorCode } = context;
+            const info = getInfo(program.getTypeChecker(), sourceFile, start, errorCode);
+            if (!info) return undefined;
+
+            if (info.kind === FixKind.MissingReturnStatement) {
+                return append(
+                    [getActionForfixAddReturnStatement(context, info.expression, info.statement)],
+                    isArrowFunction(info.declaration) ? getActionForfixRemoveBlockBodyBrace(context, info.declaration, info.expression): undefined);
+            }
+            else {
+                return [getActionForfixWrapTheBlockWithParen(context, info.declaration, info.expression)];
+            }
+        },
+        getAllCodeActions: context => codeFixAll(context, errorCodes, (changes, diag) => {
+            const info = getInfo(context.program.getTypeChecker(), diag.file, diag.start, diag.code);
+            if (!info) return undefined;
+
+            switch (context.fixId) {
+                case fixIdAddReturnStatement:
+                    addReturnStatement(changes, diag.file, info.expression, info.statement);
+                    break;
+                case fixIdRemoveBlockBodyBrace:
+                    if (!isArrowFunction(info.declaration)) return undefined;
+                    removeBlockBodyBrace(changes, diag.file, info.declaration, info.expression, /* withParen */ false);
+                    break;
+                case fixIdWrapTheBlockWithParen:
+                    if (!isArrowFunction(info.declaration)) return undefined;
+                    wrapBlockWithParen(changes, diag.file, info.declaration, info.expression);
+                    break;
+                default:
+                    Debug.fail(JSON.stringify(context.fixId));
+            }
+        }),
+    });
+
+    function updateFunctionLikeBody(declaration: FunctionLikeDeclaration, body: Block): FunctionLikeDeclaration {

```

Comment:
I think there's already a fix or refactoring for converting to/from error syntax.  I would guess it already has a helper like this.